### PR TITLE
Ditch McUser ByEmail in favour of EF-friendly method

### PR DESCRIPTION
**Pull request by timabell**
### Context

ByEmail wasn't being converted into LINQ by EF

### Changes proposed in this pull request

* Throw it away and use the newer FromSql method.
* Turn `DataServiceExportTests` into an integration test because the in-memory db used in the unit tests doesn't support `FromSql()`
* Do backflips to get the test data to be inserted only once and to obey FK and unique constraints in the real postgres database.

### CODE REVIEW

File: src/ManageCourses.Domain/Models/McOrganisationUser.cs [hunk 1]
```diff
@@ -1,5 +1,8 @@
-﻿namespace GovUk.Education.ManageCourses.Domain.Models
+﻿using System.Diagnostics;
+
+namespace GovUk.Education.ManageCourses.Domain.Models
 {
+    [DebuggerDisplay("{DebuggerDisplay,nq}")]
```
Comment (defong) [inline src/ManageCourses.Domain/Models/McOrganisationUser.cs]: Should this be here?

File: tests/ManageCourses.Tests/DbIntegration/DataServiceExportTests.cs [hunk 2]
```diff
@@ -1,47 +1,46 @@
...
     {
-        private ManageCoursesDbContext _dbContext;
         private DataService _dataService;
 
-        [OneTimeSetUp]
-        public void Setup()
+        protected override void Setup()
         {
-            _dbContext = TestHelper.GetFakeData(EnumTestType.DataService);            
+            new TestHelper(Context).BuildFakeDataForService();
```
Comment (defong) [inline tests/ManageCourses.Tests/DbIntegration/DataServiceExportTests.cs]: new it up and don't use it?

File: tests/ManageCourses.Tests/UnitTesting/Helpers/TestHelper.cs [hunk 3]
```diff
@@ -1,314 +1,305 @@
...
-                case EnumTestType.DataHelper:
-                    BuildFakeDataForHelper();
-                    break;
-            }
-            return _dbContext;
+            DbContext = dbContext;
         }
 
-        private static void SetupContext()
+        private void SetupInMemoryContext()
```
Comment (defong) [inline tests/ManageCourses.Tests/UnitTesting/Helpers/TestHelper.cs]: Is this actually getting used?

File: tests/ManageCourses.Tests/UnitTesting/Helpers/TestHelper.cs [hunk 4]
```diff
@@ -1,314 +1,305 @@
...
         {
             var optionsBuilder = new DbContextOptionsBuilder<ManageCoursesDbContext>();
             optionsBuilder.UseInMemoryDatabase("dbInMemory");
-            _dbContext = new ManageCoursesDbContext(optionsBuilder.Options);
+            DbContext = new ManageCoursesDbContext(optionsBuilder.Options);
         }
 
-        private static void BuildFakeDataForHelper()
+        public void BuildFakeDataForHelper()
```
Comment (defong) [inline tests/ManageCourses.Tests/UnitTesting/Helpers/TestHelper.cs]: Static and pass in context?

**Comment by timabell** [comment]
Okay so it was EF friendly after all. I'm just shell-shocked from having used EF6 for so long.

Going to keep the change anyway as it's generally an improvement and removes duplicate way of doing same thing.

File: src/ManageCourses.Domain/Models/McOrganisationUser.cs (hunk 1 above)
Comment (timabell) [reply src/ManageCourses.Domain/Models/McOrganisationUser.cs]: yes :-)

File: tests/ManageCourses.Tests/DbIntegration/DataServiceExportTests.cs (hunk 2 above)
Comment (timabell) [reply tests/ManageCourses.Tests/DbIntegration/DataServiceExportTests.cs]: It's used immediately by calling BuildFakeDataForService

File: tests/ManageCourses.Tests/UnitTesting/Helpers/TestHelper.cs (hunk 3 above)
Comment (timabell) [reply tests/ManageCourses.Tests/UnitTesting/Helpers/TestHelper.cs]: yes

File: tests/ManageCourses.Tests/UnitTesting/Helpers/TestHelper.cs (hunk 4 above)
Comment (timabell) [reply tests/ManageCourses.Tests/UnitTesting/Helpers/TestHelper.cs]: lots of things in this class depend on hitting the context field, I think including methods this calls.

**Pull request merged**